xcode/Subconscious/Shared/Components/Detail/Detail.swift

```diff
@@ -583,19 +505,50 @@ struct DetailEditorCursor: CursorProtocol {
     }
 }
 
+struct DetailMetaSheetCursor: CursorProtocol {
+    typealias Model = DetailModel
+    typealias ViewModel = DetailMetaSheetModel
+
+    static func get(state: Model) -> ViewModel {
+        state.metaSheet
+    }
+
+    static func set(
+        state: Model,
+        inner: ViewModel
+    ) -> Model {
+        var model = state
+        model.metaSheet = inner
+        return model
+    }
+    
+    static func tag(_ action: ViewModel.Action) -> Model.Action {
+        switch action {
+        case .selectRenameSuggestion(let suggestion):
+            return .selectRenameSuggestion(suggestion)
+        case .requestUpdateAudience(let audience):
+            return .updateAudience(audience)
+        case .requestDelete(let address):
+            return .requestDelete(address)
+        default:
+            return .metaSheet(action)
+        }
+    }
+}
+
 //  MARK: Model
 struct DetailModel: ModelProtocol {
     var address: MemoAddress?
+    var defaultAudience = Audience.local
```

Comment:
Default audience determines what kind of note will be derived (public/private) if generating an address from body content.